### PR TITLE
Add v2 version of UpdateSelf

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description

Just want a v2 version of UpdateSelf so we have complete parity.

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

**Comment by srenatus** [comment]
Unless I'm missing something, this is only about a 1-to-1 correspondence of v1 methods vs v2 methods. Is that good enough a reason to introduce the code? I think the v1 method should have never been there in the first place, and took v2 as an opportunity to fix this oddity. Functionally, there's nothing missing in v2, and there's no need for this method. Let's not introduce something that is not needed? 😉

**Comment by srenatus** [comment]
Internal previous PR with some context : https://github.com/chef/a2/pull/4791

**Comment by tylercloke** [comment]
@srenatus The functionality between `UpdateUser` and `UpdateSelf` aren't the same. For `UpdateSelf`, you need to specify your correct, previous password to make password changes. `UpdateUser` doesn't have that requirement. There is no part of the V2 API that currently supports that functionality as far as I can tell, so we need this method.

File: components/automate-gateway/api/iam/v2beta/users.proto [hunk 1]
```diff
@@ -51,4 +51,14 @@ service Users {
...
     option (chef.automate.api.iam.policy).action = "iam:users:update";
   };
+  rpc UpdateSelf (UpdateSelfReq) returns (UpdateSelfResp) {
+    option (google.api.http).put = "/iam/v2beta/self/{id}";
+    option (google.api.http).body = "*";
+    // The users namespace, as opposed for auth:users, is for actions users
+    // are allowed to perform on themselves by default without admin permissions.
+    option (chef.automate.api.policy).resource = "users:{id}";
+    option (chef.automate.api.policy).action = "update";
+    option (chef.automate.api.iam.policy).resource = "iam:usersSelf:{id}";
```
Comment (srenatus) [inline components/automate-gateway/api/iam/v2beta/users.proto]: But I still maintain that the extra resource and action is avoidable 😉

Comment (tylercloke) [reply components/automate-gateway/api/iam/v2beta/users.proto]: I'd argue we should remove the ability for users to hit UpdateUser on themselves by default.

Comment (tylercloke) [reply components/automate-gateway/api/iam/v2beta/users.proto]: Gonna merge for now so we can keep going, but feel free to refactor if you have a better approach :)

**Pull request merged**